packages/calypso-e2e/src/lib/components/support-component.ts

```diff
@@ -217,24 +233,30 @@ export class SupportComponent {
 		await this.page.click( selectors.closeButton );
 	}
 
+	/* Search input */
+
 	/**
 	 * Fills the support popover search input and waits for the query to complete.
 	 *
 	 * @param {string} text Search keyword to be entered into the search field.
 	 * @returns {Promise<void>} No return value.
 	 */
 	async search( text: string ): Promise< void > {
-		await this.page.fill( selectors.searchInput, text );
-
 		if ( text.trim() ) {
-			// If there is valid search string, then there should be a network request made
-			// resulting in a spinner. The spinner will then disappear when either the results are
-			// displayed, or no results are found.
-			await this.page.waitForSelector( selectors.spinner );
-			await this.page.waitForSelector( selectors.spinner, { state: 'hidden', timeout: 60000 } );
+			// If there is valid search string, then there should be a network request made.
+			// Wait for the response to the request and ensure the status is HTTP 200.
+			await Promise.all( [
+				this.page.waitForResponse(
+					( response ) => response.url().includes( 'search?' ) && response.status() === 200
+				),
+				this.page.fill( selectors.searchInput, text ),
+			] );
+		} else {
+			// If invalid search string (eg. '     '), then no request is made.
+			await this.page.fill( selectors.searchInput, text );
```

Comment:
Mostly out of curiosity: what was the goal in moving from the UI reflection of results having been rendered to waiting on the network response specifically? Was waiting on the spinner proving to be flaky?

Reply:
Yes, something in the Atomic + mobile viewport was causing issues with the initial steps of the test: see [TeamCity 701](https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_mobile/6534792) and [TeamCity 684](https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_mobile/6534101).

This issue went away immediately as I changed what the codebase was waiting on in [TeamCity 706](https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_mobile/6534880). 

In general, I think waiting on a network response is just as important as waiting on UI elements, and I wouldn't completely rule out the use of network inspection calls from the toolset, as long as it works on CI, local tests (against staging) and local tests (against localhost webapp).

Reply:
Well said! Sounds good to me!